core/unit_test/TestReducers.hpp

```diff
@@ -1032,6 +1032,11 @@ struct TestReducers {
     test_sum(10001);
     test_prod(35);
   }
+
+  static void execute_bool() {
+    test_LAnd(10001);
+    test_LOr(35);
```

Comment:
What about `sum`, `prod`, `max`, `min`, `minloc` and `maxloc`?

Reply:
Tests added for `sum`, `prod`, `max` and `min` pass (not pushed yet).  

```
diff --git a/core/unit_test/TestReducers.hpp b/core/unit_test/TestReducers.hpp
index 7584227..f5f2423 100644
--- a/core/unit_test/TestReducers.hpp
+++ b/core/unit_test/TestReducers.hpp
@@ -1036,6 +1036,12 @@ struct TestReducers {
   static void execute_bool() {
     test_LAnd(10001);
     test_LOr(35);
+    test_sum(10001);
+    test_prod(35);
+    test_min(10003);
+    test_max(10007);
+    test_maxloc(10007);
+    test_minloc(10003);
   }
 };
```

The `minloc` tests fails (serial and cuda) with:

```
[ RUN      ] serial.reducers_bool                                                                                                                                                             
/space/cwsmith/kokkosReductionBool/kokkos/core/unit_test/TestReducers.hpp:584: Failure                                                                                                        
Expected equality of these values:                                                                                                                                                            
  min_scalar.loc                                                                                                                                                                              
    Which is: 2147483647                                                                                                                                                                      
  reference_loc                                                                                                                                                                               
    Which is: -1                                                                                                                                                                              
[  FAILED  ] serial.reducers_bool (0 ms) 
```

IIUC, the `test_[min|max]loc` attempts to set a unique value and then confirm that it was found at the correct location.  If the `bool` scalar type can only be `true` of `false` then I wouldn't expect these tests to work.  Surprisingly, `test_maxloc` passes...  Should we exclude these tests for `bool`?

Reply:
Yes, I think it's reasonable to exclude them. Thanks for checking!

Reply:
https://github.com/kokkos/kokkos/pull/4575/commits/374aea356e4bd51ea04899f442a5b1026be8ec12 adds the tests.